src/gcoap.rs

```diff
@@ -380,7 +380,7 @@ impl<'b> PacketBuffer<'b> {
     }
 
     pub fn set_code_raw(&mut self, code: u8) {
-        unsafe { (*(*self.pkt).hdr).code = code };
+        unsafe { riot_sys::inline::coap_pkt_set_code(self.pkt as *mut coap_pkt_t as *mut riot_sys::inline::coap_pkt_t, code) };
```

Comment:
```suggestion
        unsafe { riot_sys::inline::coap_pkt_set_code(crate::inline_cast_ref(self.pkt), code) };
```

The inline_cast_* family provides some more safety around those transmutes between the "we have a C API" world and the "we have preprocessors and static inlines" world.